cli/worker.rs

```diff
@@ -50,13 +50,15 @@ impl Worker {
     let isolate = Arc::new(Mutex::new(deno::Isolate::new(startup_data, false)));
     {
       let mut i = isolate.lock().unwrap();
+      let op_registry = i.op_registry.clone();
 
       ops::compiler::init(&mut i, &state);
       ops::errors::init(&mut i, &state);
       ops::fetch::init(&mut i, &state);
       ops::files::init(&mut i, &state);
       ops::fs::init(&mut i, &state);
       ops::io::init(&mut i, &state);
+      ops::native_plugins::init(&mut i, &state, op_registry);
```

Comment:
I digged a little into the problem you described and can confirm that this line prevents workers from exiting. Unfortunately I don't have idea why. Probably because `op_registry` reference is still kept around?

Reply:
I figured it out, please apply this patch: 
```
diff --git a/cli/ops/workers.rs b/cli/ops/workers.rs
index 42f93ec5..cbc0c560 100644
--- a/cli/ops/workers.rs
+++ b/cli/ops/workers.rs
@@ -216,7 +216,12 @@ fn op_host_get_worker_closed(
   };
   let op = future.then(move |_result| {
     let mut workers_table = state_.workers.lock().unwrap();
-    workers_table.remove(&id);
+    let maybe_worker = workers_table.remove(&id);
+    if let Some(worker) = maybe_worker {
+      let mut channels = worker.state.worker_channels.lock().unwrap();
+      channels.sender.close_channel();
+      channels.receiver.close();
+    };
     futures::future::ok(json!({}))
   });
 

```